File: fijbook-alpha/app/controllers/nav/Breadcrumbs.scala
package controllers.nav

case class Breadcrumbs(trail:List[Breadcrumb]) {
  def appendChild(display:String, suffix:String): Breadcrumbs = {
    trail.headOption match {
      case Some(b)=>Breadcrumbs(Breadcrumb(display,b.link+suffix)::trail)
      case None => Breadcrumbs(List(Breadcrumb(display,suffix)))
    }
  }
  def appendChild(display:String, suffix:String, icon:String): Breadcrumbs = {
    trail.headOption match {
      case Some(b)=>Breadcrumbs(Breadcrumb(display,b.link+suffix, Some(icon))::trail)
      case None => Breadcrumbs(List(Breadcrumb(display,suffix, Some(icon))))
    }
  }
  def append(display:String, link:String): Breadcrumbs = {
    copy(trail=Breadcrumb(display,link)::trail)
  }
  def append(display:String, link:String, icon:String): Breadcrumbs = {
    copy(trail=Breadcrumb(display,link,Some(icon))::trail)
  }

}

object Breadcrumbs {
  def base=Breadcrumbs(List(Breadcrumb("Deep Fij", "/deepfij", Some("fa-circle-o"))))
}

case class Breadcrumb(display:String, link:String, icon:Option[String]=None)

File: fijbook-alpha/public/javascripts/components/Breadcrumbs.js
'use strict';

class Breadcrumbs extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return <ol className="breadcrumb">
            {
                this.props.crumbs.map(c => <li><i className="fa"/><a href={c.link}>{c.display}</a></li>)
            }
        </ol>;
    }
}

const crumbs = [
    {
        display: "Deep Fij",
        link: "/deepfij",
        fontAwesomeIcon: "fa-circle-o"
    },
    {
        display: "Teams",
        link: "/deepfij/teams",
        fontAwesomeIcon: "fa-users"
    },
    {
        display: "Appalacian St.",
        link: "/deepfij/teams/appaliachian-st",
        fontAwesomeIcon: ""
    },
];

ReactDOM.render(<Breadcrumbs crumbs={crumbs}/>, document.querySelector('#breadcrumbs'));
